### PR TITLE
Allow to opt-out from clearing cache on shutdown

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

**Comment by Nyholm** [comment]
We could also do a protected property

**Comment by chapterjason** [comment]
@Nyholm I think it would be better to create a configurable Property as you suggested, this would end up with a better DX. The shutdown will be called automatically from the [KernelTestCase](https://github.com/symfony/symfony/blob/174c92eefb1b29a6bd2ca73cc9af6e50b9cb4b55/src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php#L146)

**Comment by Nyholm** [comment]
I've updated the PR.

File: src/TestKernel.php [hunk 1]
```diff
@@ -181,10 +188,14 @@ protected function configureRoutes($routes): void
         }
     }
 
-    public function shutdown(): void
+    public function shutdown($clearCache = true): void
```
Comment (jdreesen) [inline src/TestKernel.php]: As far as I can see, this isn't used anymore:

```suggestion
    public function shutdown(): void
```

Comment (Nyholm) [reply src/TestKernel.php]: Oops. Thank you.

**Review by chapterjason — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**